src/test/java/com/puppycrawl/tools/checkstyle/checks/imports/UnusedImportsCheckTest.java

```diff
@@ -249,4 +249,15 @@ public void testImportsJavadocQualifiedName() throws Exception {
         verify(checkConfig, getPath("InputUnusedImportsJavadocQualifiedName.java"), expected);
     }
 
+    @Test
+    public void testInnerInterfacesAndClasses() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(UnusedImportsCheck.class);
+        final String[] expected = {
+            "3:8: " + getCheckMessage(MSG_KEY, "java.util.Set"),
+            "4:8: " + getCheckMessage(MSG_KEY, "java.util.List"),
+            "5:8: " + getCheckMessage(MSG_KEY, "java.util.Map"),
+
+        };
+        verify(checkConfig, getPath("InputUnusedImportsInner.java"), expected);
+    }
```

Comment:
please add to test cases the only issue that was found during regression, it is different from new one because the violation is on top level class
https://gaurabdg.github.io/checkstyle-tester-reports/regression/UnusedImports/diff/openjdk7/xref/home/gaurabdg/Workspace/contribution/checkstyle-tester/repositories/openjdk7/src/share/classes/sun/java2d/cmm/lcms/LCMS.java.html#L33

Reply:
Done.

Reply:
Please make it a separate test case to avoid confusion

Reply:
Done